File: code-smells-project/src/middlewares/error_handler.py
"""Tratamento de erro centralizado — sem vazar stack trace ao cliente."""
from flask import jsonify
from src.middlewares.validation import ValidationError


def register_error_handlers(app):
    @app.errorhandler(ValidationError)
    def handle_validation(err):
        return jsonify({"erro": err.mensagem, "sucesso": False}), err.status

    @app.errorhandler(404)
    def handle_not_found(_err):
        return jsonify({"erro": "Recurso não encontrado", "sucesso": False}), 404

    @app.errorhandler(Exception)
    def handle_unexpected(_err):
        # Não expõe detalhes internos ao cliente.
        return jsonify({"erro": "Erro interno do servidor", "sucesso": False}), 500

File: ecommerce-api-legacy/src/middlewares/errorHandler.js
'use strict';

// Tratamento de erro centralizado — não expõe stack trace ao cliente.
// eslint-disable-next-line no-unused-vars
function errorHandler(err, req, res, next) {
    const status = err.status || 500;
    const message = err.status ? err.message : 'Erro interno do servidor';
    res.status(status).send(message);
}

module.exports = { errorHandler };
